### ISSUE TITLE
Update install instructions in the docs for OSX

### ISSUE THREAD

**Issue by mastoj** [typelead/eta#236]
I tried to install eta but failed. I think for OSX the prerequisite list should be updated to include xcode, but it might be a "smaller" prerequisite you can use. Just mentioning Stack and java is definitely not enough.

I installed xcode which took me one step further, but the install still fails for me:

```
Preparing to install GHC to an isolated location.
This will not interfere with any system-level installation.
Already downloaded.
Running /Users/tomasjansson/.stack/programs/x86_64-osx/ghc-7.10.3.temp/ghc-7.10.3/configure --prefix=/Users/tomasjansson/.stack/programs/x86_64-osx/ghc-7.10.3/ in directory /Users/tomasjansson/.stack/programs/x86_64-osx/ghc-7.10.3.temp/ghc-7.10.3/ exited with ExitFailure 77

checking for path to top of build tree... /Users/tomasjansson/.stack/programs/x86_64-osx/ghc-7.10.3.temp/ghc-7.10.3
Build platform inferred as: x86_64-apple-darwin
Host platform inferred as: x86_64-apple-darwin
Target platform inferred as: x86_64-apple-darwin
GHC build  : x86_64-apple-darwin
GHC host   : x86_64-apple-darwin
GHC target : x86_64-apple-darwin
checking for perl... /usr/bin/perl
checking for a BSD-compatible install... /usr/local/bin/ginstall -c
checking whether ln -s works... yes
checking for gsed... sed
checking XCode version... 8.2.1
configure: XCode version component 1: 8
configure: XCode version component 2: 2
checking for gcc... /usr/bin/gcc
checking for llc-3.5... no
checking for llc... /Library/Frameworks/Mono.framework/Versions/Current/bin//llc
checking /Library/Frameworks/Mono.framework/Versions/Current/bin//llc is version 3.5... no
checking for opt-3.5... no
checking for opt... /Library/Frameworks/Mono.framework/Versions/Current/bin//opt
checking /Library/Frameworks/Mono.framework/Versions/Current/bin//opt is version 3.5... no
checking for ld... /usr/bin/ld
checking for gcc... /usr/bin/gcc
checking whether the C compiler works... no

configure: WARNING: configure can't recognize your CPP program, you may need to set --with-hs-cpp-flags=FLAGS explicitly
configure: error: in `/Users/tomasjansson/.stack/programs/x86_64-osx/ghc-7.10.3.temp/ghc-7.10.3':
configure: error: C compiler cannot create executables
See `config.log' for more details
```

Let me know if there is any other information I you want.

**Comment by rahulmutt** [comment]
Yes, XCode is definitely a requirement as that seems to be the easiest way to get the standard build tools. I've updated the docs (changes should be visible soon). Thanks for pointing that out.

As for the other errors, have you tried looking at the docs for the Stack build tool? It seems to be an error related to that. How have you installed it? 

Can you run the following commands and share the output?
```
$ cpp --version
$ gcc --version
```

**Comment by mastoj** [comment]
Installed stack with the command on the stack page: 

    curl -sSL https://get.haskellstack.org/ | sh

I found the problem though, and it hasn't to do with `cpp` or `gcc`. Things I needed to do, which might be good to have in a troubleshoot section or something, are:

* Accept xcode license and such, you will get prompted on this when running `cpp --version`
* Make sure the folder `~/.local/bin~` is added to the path and it exists. 

That made everything compile and run as expected.
